packages/teleport/src/components/RecoveryCodes/RecoveryCodes.tsx

```diff
@@ -32,6 +32,15 @@ export default function RecoveryCodesDialog({
 }: Props) {
   const codesRef = useRef();
 
+  const captureRecoveryCodeEvent = (event: CaptureEvent) => {
+    if (username) {
+      userEventService.capturePreUserEvent({
+        event: event,
+        username: username,
+      });
```

Comment:
i don't remember if we discussed this before, but components that live in `src/components` are normally shared. I believe this component is used in the following flows:

1) on reset or on invite
2) user is logged in, goes to their account settings, and goes to their recovery code section and clicks on `generate new recovery codes` (so this is not a `preUser` event.)

Not sure if that detail mattered

Reply:
Yep! Same [story](https://github.com/gravitational/webapps/pull/1344#discussion_r1038570418) as in #1344 

This is fine, we may add more events or logic to this component when we instrument other aspects of UI interaction.